pkg/providers/v1/aws.go

```diff
@@ -5130,3 +5130,34 @@ func (c *Cloud) describeNetworkInterfaces(nodeName string) (*ec2.NetworkInterfac
 	}
 	return eni.NetworkInterfaces[0], nil
 }
+
+type regionDetails struct {
+	regionName string
+	zone       string
+}
+
+func getRegionFromMetadata(cfg CloudConfig, metadata EC2Metadata) (*regionDetails, error) {
```

Comment:
consider returning multiple return values like
```
(string, string, err)
region, zone, err 
```
instead. I like it better than a special struct